src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -44,7 +65,12 @@ protected MessagePump(IConfiguration configuration, IServiceProvider serviceProv
         /// <summary>
         /// Gets the boolean flag that indicates whether the message pump is started and receiving messages.
         /// </summary>
-        public bool IsStarted { get; private set; }
+        public bool IsStarted { get; protected set; }
+
+        /// <summary>
+        /// Gets the current state of the message pump within the circuit breaker context.
+        /// </summary>
+        protected MessagePumpCircuitState CircuitState { get; set; } = MessagePumpCircuitState.Closed;
```

Comment:
The service bus message pump sets this, but we might be able to make this private indeed.

Reply:
If it should only be set via a set of methods, I'd make this private.
What would happen if an inheritor sets this property directly ?

Reply:
The service bus message pump needs this, I believe, to activate it back again. I believe that was also the reason it was made `protected`, so that client implementations of the message pump have control. The `MessagePump` does not contain any functionality.

Reply:
To clarify: the `MessagePump` does not have any functionality and acts here more as an interface than as the parent class. It provides the necessary properties and start/stop information, but not the actual implementation. What it does help us with, is streamlining message pumps and adding functionality to pumps without knowing its implementation.

The circuit breaker state here is only for the implementations to use, and in this case the half-open state is something that this interface does not know where to set.